tools/distpartitioning/array_readwriter/parquet.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+import pandas as pd
+import pyarrow
+import pyarrow.parquet
+
+from .registry import register_array_parser
+
+
+@register_array_parser("parquet")
+class ParquetArrayParser(object):
+    def __init__(self):
+        pass
+
+    def read(self, path):
+        logging.info("Reading from %s using parquet format" % path)
+        metadata = pyarrow.parquet.read_metadata(path)
+        metadata = metadata.schema.to_arrow_schema().metadata
+        # As parquet data are tabularized, we assume the dim of ndarray is 2.
+        # If not, it should be explictly specified in the file as metadata.
+        shape = metadata.get(b"shape", None)
+        table = pyarrow.parquet.read_table(path, memory_map=True)
+        logging.info("Done reading from %s" % path)
+        arr = table.to_pandas().to_numpy()
+        if not shape:
+            logging.warning(
+                "Shape info not found in the metadata,"
```

Comment:
Make full use of the 80 chars before change to new line.

Reply:
fixed